### PR TITLE
Add table of contents to readme

### PR DESCRIPTION
**Pull request by mikaelarguedas**
In an attempt to make the page more user-friendly for people reading it on dockerhub
Related to #358 and #186

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,15 @@
...
 This repo hosts Dockerfiles and scaffolding for images in the Official Library and OSRF Organization on Docker Hub.
 
+## Table of Contents
+[Official docker library](#official-library)<br/>
+[Official ROS (1 and 2) images](#ros--)<br/>
+[Official Gazebo images](#gazebo--)<br/>
+[OSRF ROS (1 and 2) overlay images](#ros---1)<br/>
+[OSRF ROS 2 development images](#ros2--)<br/>
+[OSRF Gazebo overlay images](#gazebo---1)<br/>
+[OSRF ROS Legacy images](#ros-legacy--)<br/>
```
Comment (mikaelarguedas) [inline README.md]: Note to self: these will need to be made absolute before merging otherwise I think it will not work on dockerhub

**Comment by ruffsl** [comment]
Perhaps we could just point out/link to the tags page on the respective docker hub repo, rather than hardcoding what tags do or do not yet exist? I think educating docker users about the conventions of latest is out of scope for top level readme.

**Comment by mikaelarguedas** [comment]
Yeah that was my original feeling as well.

As people seem to be confused by the lack of latest (and most popular dockerhub repos have a latest) I figured maybe we were enough outside of the "default" to justify explaining it after all.
But just mentionning the command format and point to the tag list may be enough :+1:

How about something like this:
"""
Images must be pulled using an explicit tag e.g:
```
docker pull osrf/ros:<tag_name>
```
List of tags available at https://hub.docker.com/r/osrf/ros/tags
"""

**Comment by ruffsl** [comment]
That looks good.

**Pull request merged**